### PR TITLE
Latest SDK generation for Guest configuration with updated tests

### PR DESCRIPTION
**Pull request by anineela**
Our latest spec is already merged to main: https://github.com/Azure/azure-rest-api-specs/tree/main/specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2021-01-25

Test success:
image

Build success:
image

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/src/Generated/Models/GuestConfigurationAssignmentProperties.cs [hunk 1]
```diff
@@ -55,13 +57,14 @@ public GuestConfigurationAssignmentProperties()
...
+        public GuestConfigurationAssignmentProperties(string targetResourceId = default(string), GuestConfigurationNavigation guestConfiguration = default(GuestConfigurationNavigation), string complianceStatus = default(string), System.DateTime? lastComplianceStatusChecked = default(System.DateTime?), string latestReportId = default(string), string parameterHash = default(string), AssignmentReport latestAssignmentReport = default(AssignmentReport), string context = default(string), string assignmentHash = default(string), string provisioningState = default(string), string resourceType = default(string), IList<VMSSVMInfo> vmssVMList = default(IList<VMSSVMInfo>))
```
Comment (ArthurMa1978) [inline sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/src/Generated/Models/GuestConfigurationAssignmentProperties.cs:60]: As interface changes caused not backward compatible, please increate the major version of the package, not minor version

File: sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/src/Generated/Models/GuestConfigurationNavigation.cs [hunk 2]
```diff
@@ -44,18 +44,30 @@ public GuestConfigurationNavigation()
...
+        /// configuration parameters for the guest configuration.</param>
         /// <param name="configurationSetting">The configuration setting for
         /// the guest configuration.</param>
-        public GuestConfigurationNavigation(string kind = default(string), string name = default(string), string version = default(string), string contentUri = default(string), string contentHash = default(string), IList<ConfigurationParameter> configurationParameter = default(IList<ConfigurationParameter>), ConfigurationSetting configurationSetting = default(ConfigurationSetting))
+        public GuestConfigurationNavigation(string kind = default(string), string name = default(string), string version = default(string), string contentUri = default(string), string contentHash = default(string), string assignmentType = default(string), string contentType = default(string), IList<ConfigurationParameter> configurationParameter = default(IList<ConfigurationParameter>), IList<ConfigurationParameter> configurationProtectedParameter = default(IList<ConfigurationParameter>), ConfigurationSetting configurationSetting = default(ConfigurationSetting))
```
Comment (ArthurMa1978) [inline sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/src/Generated/Models/GuestConfigurationNavigation.cs:60]: As interface changes caused not backward compatible, please increate the major version of the package, not minor version

File: sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/src/Microsoft.Azure.Management.GuestConfiguration.csproj [hunk 3]
```diff
@@ -6,7 +6,7 @@
   <PropertyGroup>
     <PackageId>Microsoft.Azure.Management.GuestConfiguration</PackageId>
     <Description>Provides Microsoft Azure Guest Configuration management operations including the ability to create, update and get guest configuration assignments, compliance statuses, guest configuration assignment reports.</Description>
-    <Version>1.5.0</Version>
+    <Version>1.6.0</Version>
```
Comment (ArthurMa1978) [inline sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/src/Microsoft.Azure.Management.GuestConfiguration.csproj:9]: As interface changes caused not backward compatible, please increate the major version of the package, not minor version

File: sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/src/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -7,8 +7,8 @@
 [assembly: AssemblyTitle("Microsoft Azure GuestConfiguration Management Library")]
 [assembly: AssemblyDescription("Provides Microsoft Azure Guest Configuration management operations including the ability to create, update and get guest configuration assignments, compliance statuses, guest configuration assignment reports.")]
 
-[assembly: AssemblyVersion("1.5.0")]
-[assembly: AssemblyFileVersion("1.5.0")]
+[assembly: AssemblyVersion("1.6.0")]
```
Comment (ArthurMa1978) [inline sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/src/Properties/AssemblyInfo.cs:10]: As interface changes caused not backward compatible, please increate the major version of the package, not minor version

File: sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/src/Generated/Models/GuestConfigurationAssignmentProperties.cs (hunk 1 above)
Comment (anineela) [reply sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/src/Generated/Models/GuestConfigurationAssignmentProperties.cs:60]: done

File: sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/src/Microsoft.Azure.Management.GuestConfiguration.csproj (hunk 3 above)
Comment (anineela) [reply sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/src/Microsoft.Azure.Management.GuestConfiguration.csproj]: done

File: sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/src/Properties/AssemblyInfo.cs (hunk 4 above)
Comment (anineela) [reply sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/src/Properties/AssemblyInfo.cs]: done

File: sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/src/Generated/Models/GuestConfigurationNavigation.cs (hunk 2 above)
Comment (anineela) [reply sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/src/Generated/Models/GuestConfigurationNavigation.cs:60]: sure. done

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**